File: Questão7.go
package main

import "fmt"


type Escola struct {
    Nome       string
    Professores []Professor
}


func (e *Escola) AdicionarProfessor(p Professor) {
    e.Professores = append(e.Professores, p)
}


type Professor struct {
    Nome string
}

func main() {
    escola := Escola{Nome: "Escola Primária"}
    prof1 := Professor{Nome: "Alice"}
    prof2 := Professor{Nome: "Bob"}
    
    escola.AdicionarProfessor(prof1)
    escola.AdicionarProfessor(prof2)
    
    fmt.Println("Professores da", escola.Nome)
    for _, professor := range escola.Professores {
        fmt.Println(professor.Nome)
    }
}

File: Questão7.java
import java.util.ArrayList;
import java.util.List;

class Escola {
    private String nome;
    private List<Professor> professores;

    public Escola(String nome) {
        this.nome = nome;
        this.professores = new ArrayList<>();
    }

    public void adicionarProfessor(Professor professor) {
        professores.add(professor);
    }

    public String getNome() {
        return nome;
    }

    public List<Professor> getProfessores() {
        return professores;
    }
}

class Professor {
    private String nome;

    public Professor(String nome) {
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }
}

public class Main {
    public static void main(String[] args) {
        Escola escola = new Escola("Escola Primária");
        Professor prof1 = new Professor("Alice");
        Professor prof2 = new Professor("Bob");
        
        escola.adicionarProfessor(prof1);
        escola.adicionarProfessor(prof2);
        
        System.out.println("Professores da " + escola.getNome() + ":");
        for (Professor professor : escola.getProfessores()) {
            System.out.println(professor.getNome());
        }
    }
}
